### PR TITLE
[AC-7327] Update directory Entrepreneur profile pages with accurate labels

### PR DESCRIPTION
**Pull request by PrinceDavis**
Changes introduced in [AC-7327](https://masschallenge.atlassian.net/browse/AC-7327):
Update directory Entrepreneur profile pages with accurate labels

**Test instructions:**
- See related PR below on how to test the interface

**Updates**
#### This attempts to 
- Get a users program role
- Filter those that doesn't correspond to `an entrepreneur` participation in a program
- Group the result by program family

@shanbady, is there any other way of getting an entrepreneur previous and current program participation that is more efficient?

[related to](https://github.com/masschallenge/front-end/pull/215)

### CODE REVIEW

**Comment by freshprincekla** [comment]
Need some tests to reflect changes you have made.

**Comment by PrinceDavis** [comment]
@iankigen this is ready for review

File: web/impact/impact/tests/test_graphql.py
Comment (iankigen) [inline web/impact/impact/tests/test_graphql.py]: How about you import and  use `get_user_prg_by_programfamily` instead

File: web/impact/impact/graphql/types/expert_profile_type.py
Comment (shanbady) [inline web/impact/impact/graphql/types/expert_profile_type.py:130]: seems like this is a duplicate method. can we extract this out to a shared function?

**Review by shanbady — DISMISSED** [review]

File: web/impact/impact/graphql/types/expert_profile_type.py
Comment (PrinceDavis) [reply web/impact/impact/graphql/types/expert_profile_type.py:130]: @shanbady what other function does this duplicate?

File: web/impact/impact/graphql/types/entrepreneur_profile_type.py
Comment (iankigen) [inline web/impact/impact/graphql/types/entrepreneur_profile_type.py]: remove the commented print statements

File: web/impact/impact/graphql/types/expert_profile_type.py
Comment (shanbady) [reply web/impact/impact/graphql/types/expert_profile_type.py:130]: resolve_program_roles shows up here:

https://github.com/masschallenge/impact-api/pull/378/files/9f26df4253eadcf28a4c6d2f87547a36025ba953#diff-d0b1a8955f0c1743915a8a2c97703158R63
and here:
https://github.com/masschallenge/impact-api/pull/378/files/9f26df4253eadcf28a4c6d2f87547a36025ba953#diff-2807098ef5f50df497dcaf6d1591b111R116

File: web/impact/impact/utils.py
Comment (iankigen) [inline web/impact/impact/utils.py]: consider refactoring this to one function. Similar block also found on `def get_user_prg_role_by_program_family`

Comment (PrinceDavis) [reply web/impact/impact/utils.py]: @iankigen I am not sure what you are referring to here, please say more

Comment (iankigen) [reply web/impact/impact/utils.py]: [This block](https://github.com/masschallenge/impact-api/blob/b9c74bd88050804b17e17826d4532b279ecb39a8/web/impact/impact/utils.py#L88-L92) has similar functionality to [this](https://github.com/masschallenge/impact-api/blob/b9c74bd88050804b17e17826d4532b279ecb39a8/web/impact/impact/utils.py#L117-L121). Consider refactoring to one helper function

Comment (PrinceDavis) [reply web/impact/impact/utils.py]: I don't really see the additional benefit in abstracting those five lines, I think leaving it the way it is helps readability.

However I have done further abstraction on the query level

Comment (PrinceDavis) [reply web/impact/impact/utils.py]: @shanbady  @iankigen This is ready for final review

**Review by iankigen — APPROVED** [review]
LGTM

File: web/impact/impact/graphql/types/expert_profile_type.py
Comment (geofrocker) [reply web/impact/impact/graphql/types/expert_profile_type.py:130]: @PrinceDavis This needs to resolved

Comment (PrinceDavis) [reply web/impact/impact/graphql/types/expert_profile_type.py:130]: This has been resolved

Comment (geofrocker) [reply web/impact/impact/graphql/types/expert_profile_type.py:130]: Thanks for the clarification

**Comment by iankigen** [comment]
Resolve conflicts

**Comment by geofrocker** [comment]
While testing with TG, I experienced intermittent frontend errors on local. It would be great to first test this on a test machine before merging.

**Review by geofrocker — COMMENTED** [review]
While testing with TG, I experienced intermittent frontend errors on local. It would be great to first test this on a test machine before merging.

**Review by geofrocker — APPROVED** [review]
Thanks TG. Code and QA look good.
- I have confirmed that api is able to serve the data required for the frontend to work
- Entrepreneurship history and Participation details can be utilised by the front-end

**Comment by geofrocker** [comment]
Do not merge until the sibling PR on frontend is approved. 🚨🚨

File: web/impact/impact/graphql/types/expert_profile_type.py
Comment (jonkiparsky) [reply web/impact/impact/graphql/types/expert_profile_type.py:130]: FWIW, code climate thinks this hasn't been resolved (still showing a duplication warning)

File: web/impact/impact/tests/test_graphql.py
Comment (jonkiparsky) [inline web/impact/impact/tests/test_graphql.py]: style note: imports should be on separate lines (and sorted alphabetically)

Comment (PrinceDavis) [reply web/impact/impact/tests/test_graphql.py]: Taken care of

**Pull request merged**
